Lucid/Core/CoreManager.swift

```diff
@@ -684,6 +933,202 @@ private extension CoreManager {
+
+                                        let entitiesToUpdate = self.filter(entities: remoteResults, basedOn: time).compactMap { $0 }
+                                        guard entitiesToUpdate.isEmpty == false else {
+                                            continuation.resume(returning: remoteResults)
+                                            return
+                                        }
+                                        self.setUpdateTime(time, for: entitiesToUpdate.lazy.map { $0.identifier })
+
+                                        self.localStore.set(entitiesToUpdate, in: WriteContext(dataTarget: .local)) { result in
+                                            if let error = result?.error {
+                                                Logger.log(.error, "\(CoreManager.self): An error occurred while writing entities: \(error)", assert: true)
+                                            }
+                                            continuation.resume(returning: remoteResults)
+                                            self.raiseUpdateEvents(withQuery: query,
+                                                                   results: remoteResults,
+                                                                   returnsCompleteResultSet: context.returnsCompleteResultSet)
+                                        }
+                                    }
+                                }
+                            } else {
+                                continuation.resume(returning: remoteResults)
+                                property.update(with: remoteResults)
+                            }
+
+                        case .failure(let error):
+                            if error.shouldFallBackToLocalStore {
+                                Logger.log(.debug, "\(CoreManager.self): Did not receive data while searching entities: \(error). Will fall back to local store if possible.")
+                            } else {
+                                Logger.log(.error, "\(CoreManager.self): An error occurred while searching entities: \(error)", assert: true)
+                            }
+                            continuation.resume(throwing: ManagerError.store(error))
+                        }
+                    }
+                }
+            }
+
+            return result
+        }
+
+        let continuous = AsyncStream<QueryResult<E>>() { continuation in
+            property
+                .sink(receiveCompletion: { _ in
+                    continuation.finish()
+                }, receiveValue: { value in
+                    continuation.yield(value)
+                })
+                .store(in: cancellable)
+        }
+
+        guard context.responseAllowedForAccessLevel,
+            initialUserAccess == context.userAccess else {
+            throw ManagerError.userAccessInvalid
+        }
```

Comment:
I'm confused, this looks like the same code that was failing from last Friday.

I would think we want to move this whole guard statement up to line `1108` to check before we return the result.